DataStructures/Trees/BinaryTree.cs

```diff
@@ -6,7 +6,17 @@
 
 namespace DataStructures.Trees
 {
-    class BinaryTree
+    public class BinaryTree : BinarySearchTree
```

Comment:
This is actually backwards - this class with `Add`/`Contains` should be named `BinarySearchTree`, and inherit from `BinaryTree` with the traversals (and Max and BreadthFirst).